staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3562,7 +3562,16 @@ type PodSecurityContext struct {
 	RunAsNonRoot *bool `json:"runAsNonRoot,omitempty" protobuf:"varint,3,opt,name=runAsNonRoot"`
 	// A list of groups applied to the first process run in each container, in addition
 	// to the container's primary GID.  If unspecified, no groups will be added to
-	// any container.
+	// any container. Please remember that the group membership defined in the
+	// container image is kept. That means, if
+	// - `runAsUser:1000, runAsGroup:1000, supplementalGroups:[60000]` is specified in spec,
+	// - and `uid=1000` belongs to `gid=50000` in the container image,
+	// then the first process in the container belongs to
+	// `groups=1000,50000(defined in the image),60000(defined in supplementalGroups)`.
+	// Thus, even if the cluster enforces some security policy that protects the value of
+	// supplementalGroups field, the effect of its enforcement is limited. Unexpected
+	// group membership will cause unexpected file access permissions, particularly
+	// when using hostPath volumes.
```

Comment:
I wonder if this example / hostPath reference are too narrowly focused for this field documentation, and if we could be just as clear with a more general statement like https://github.com/kubernetes/kubernetes/pull/113047/files#r995797820

Reply:
> I wonder if this example / hostPath reference are too narrowly focused for this field documentation

I agree. Your statement is general and quite simple enough.